From be40ad882d16edd38059ca492668b6146025ec20 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Mon, 26 Jul 2021 15:53:11 +0300
Subject: [PATCH] [SW-45031] greco: avoid asic specific duplications

In this patch, Greco HL_TARGET asic specific
common code moved to common dir.

Change-Id: I02abb91d1f6e58987c63f12684be228fae1ad1ed
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/greco/arc/asic.c b/arch/greco/arc/asic.c
index 19a576a..6557614 100644
--- a/arch/greco/arc/asic.c
+++ b/arch/greco/arc/asic.c
@@ -1,125 +1,14 @@
+// SPDX-License-Identifier:     GPL-2.0+
 /*
  * (C) Copyright 2019 HabanaLabs, Ltd.
  * Noam Camus <noamikong@gmail.com>
  *
- * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <psoc_global_conf_regs.h>
-#include <psoc_gpio0_regs.h>
-#include <cpu_if_regs.h>
 
-#include <string.h>
-#include <bitops.h>
-#include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-#include <boot_desc.h>
-#include <hl_boot_if.h>
-#include <pci.h>
-
-#include "common.h"
-
-u32 hl_skip_pcie_init(void)
-{
-	/* Use DumpSel[2] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) & BIT(10);
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE, HL_PLL_DIV3);
-}
-
-void hl_uart_init(void)
-{
-	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
-						HL_PLL_DIV3);
-
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
-}
 
 /* configure yellow LED(GPIO2) in output mode and turn on(active low) */
 void preboot_led_on(void)
 {
 	/* TODO: perform LED GPIO configuration */
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor;
-	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
-						HL_PLL_DIV3);
-
-	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
-}
-
-void hl_pci_iatu_init(void)
-{
-	struct hl_pci_inbound_region inbound_region;
-	struct hl_pci_outbound_region outbound_region;
-
-	hl_pcie_disable_access();
-
-	/* Inbound Region 0 - Bar - - Point to SRAM and CFG */
-	inbound_region.mode = PCI_BAR_MATCH_MODE;
-	inbound_region.bar = PCI_BAR_ID_0;
-	inbound_region.addr = SDRAM_FULL_BASE;
-	hl_pci_set_inbound_region(0, &inbound_region);
-
-	/* Inbound Region 1 - Bar 4 - Point to DRAM */
-	inbound_region.mode = PCI_BAR_MATCH_MODE;
-	inbound_region.bar = PCI_BAR_ID_4;
-	inbound_region.addr = DRAM_BASE;
-	hl_pci_set_inbound_region(1, &inbound_region);
-
-	/* Outbound Region 0 - Point to Host */
-	outbound_region.addr = HOST_PHYS_BASE;
-	outbound_region.size = HOST_PHYS_SIZE;
-	hl_pci_set_outbound_region(&outbound_region);
-
-	hl_pcie_enable_access();
-}
-
-void lkd_dev_spec_regs(struct cpu_dyn_regs *regs)
-{
-	regs->cpu_pq_base_addr_low = mmCPU_IF_PQ_BASE_ADDR_LOW;
-	regs->cpu_pq_base_addr_high = mmCPU_IF_PQ_BASE_ADDR_HIGH;
-	regs->cpu_pq_length = mmCPU_IF_PQ_LENGTH;
-	regs->cpu_pq_init_status = mmCPU_IF_QUEUE_INIT;
-	regs->cpu_eq_base_addr_low = mmCPU_IF_EQ_BASE_ADDR_LOW;
-	regs->cpu_eq_base_addr_high = mmCPU_IF_EQ_BASE_ADDR_HIGH;
-	regs->cpu_eq_length = mmCPU_IF_EQ_LENGTH;
-	regs->cpu_eq_ci = mmCPU_IF_EQ_RD_OFFS;
-	regs->cpu_cq_base_addr_low = mmCPU_IF_CQ_BASE_ADDR_LOW;
-	regs->cpu_cq_base_addr_high = mmCPU_IF_CQ_BASE_ADDR_HIGH;
-	regs->cpu_cq_length = mmCPU_IF_CQ_LENGTH;
-	regs->cpu_pf_pq_pi = mmCPU_IF_PF_PQ_PI;
-	regs->cpu_boot_dev_sts0 = mmCPU_BOOT_DEV_STS0;
-	regs->cpu_boot_dev_sts1 = mmCPU_BOOT_DEV_STS1;
-	regs->cpu_boot_err0 = mmCPU_BOOT_ERR0;
-	regs->cpu_boot_err1 = mmCPU_BOOT_ERR1;
-	regs->cpu_boot_status = mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
-	regs->fw_upd_sts = mmUPD_STS;
-	regs->fw_upd_cmd = mmUPD_CMD;
-	regs->fw_upd_pending_sts = mmUPD_PENDING_STS;
-	regs->fuse_ver_offset = mmFUSE_VER_OFFSET;
-	regs->preboot_ver_offset = mmPREBOOT_VER_OFFSET;
-	regs->uboot_ver_offset = mmUBOOT_VER_OFFSET;
-	regs->hw_state = mmHW_STATE;
-	regs->kmd_msg_to_cpu = mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU;
-	regs->cpu_cmd_status_to_host = mmCPU_CMD_STATUS_TO_HOST;
-	regs->gic_tpc_qm_irq_ctrl = mmGIC_TPC_QM_IRQ_CTRL_POLL_REG;
-	regs->gic_mme_qm_irq_ctrl = mmGIC_MME_QM_IRQ_CTRL_POLL_REG;
-	regs->gic_dma_qm_irq_ctrl = mmGIC_DMA_QM_IRQ_CTRL_POLL_REG;
-	regs->gic_rot_qm_irq_ctrl = mmGIC_ROT_QM_IRQ_CTRL_POLL_REG;
-	regs->gic_dma_core_irq_ctrl = mmGIC_DMA_CR_IRQ_CTRL_POLL_REG;
-	regs->gic_host_pi_upd_irq = mmGIC_HOST_PI_UPD_IRQ_POLL_REG;
-	regs->gic_host_halt_irq = mmGIC_HOST_HALT_IRQ_POLL_REG;
-	regs->gic_host_ints_irq  = mmGIC_HOST_INTS_IRQ_POLL_REG;
-	regs->gic_host_soft_rst_irq = mmGIC_HOST_SOFT_RST_IRQ_POLL_REG;
 }
diff --git a/arch/greco/arc_elf32/asic.c b/arch/greco/arc_elf32/asic.c
index c2f48c0..bcb4aba 100644
--- a/arch/greco/arc_elf32/asic.c
+++ b/arch/greco/arc_elf32/asic.c
@@ -8,42 +8,8 @@
 
 #include <bitops.h>
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
 
 #include "common.h"
-
-u32 hl_skip_pcie_init(void)
-{
-	/* Use DumpSel[2] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) & BIT(10);
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE, HL_PLL_DIV3);
-}
-
-void hl_uart_init(void)
-{
-	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
-						HL_PLL_DIV3);
-
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor;
-	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
-						HL_PLL_DIV3);
-
-	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
-}
 
 /*
  * FW_SRAM layout:
diff --git a/arch/greco/arc_elf32/common.h b/arch/greco/arc_elf32/common.h
index d7a67c2..b13c7b6 100644
--- a/arch/greco/arc_elf32/common.h
+++ b/arch/greco/arc_elf32/common.h
@@ -18,6 +18,10 @@
 
 #define SDRAM_BASE		0x10000000
 #define SDRAM_FULL_BASE		0x7FF0000000ull
+#define DRAM_BASE		0x1000000000000ull
+#define HOST_PHYS_BASE		0x2000000000000ull
+#define HOST_PHYS_SIZE		0x2000000000000ull
+
 #define HL_SP_SRAM_BASE		0x7FFBFE0000ULL
 #define PCIE_FW_SRAM_BASE	0x1BFF0000
 #define FW_SRAM_BASE		0x1B050000
diff --git a/arch/greco/arm/asic.c b/arch/greco/arm/asic.c
index 888fbf2..7570783 100644
--- a/arch/greco/arm/asic.c
+++ b/arch/greco/arm/asic.c
@@ -7,7 +7,6 @@
 
 #include <psoc_global_conf_regs.h>
 #include <psoc_gpio0_regs.h>
-#include <cpu_if_regs.h>
 
 #include <string.h>
 #include <bitops.h>
@@ -16,45 +15,8 @@
 #include <uart.h>
 #include <sha256.h>
 #include <boot_desc.h>
-#include <hl_boot_if.h>
-#include <pci.h>
 
 #include "common.h"
-
-u32 hl_skip_pcie_init(void)
-{
-	/* Use DumpSel[2] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) & BIT(10);
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	/* As per Goya2 PLL cfg.xslx, PCI PLL Div[3] drives CPU timestamp */
-	return hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE, HL_PLL_DIV3);
-}
-
-void hl_uart_init(void)
-{
-	/* As per Goya2 PLL cfg.xlsx, PCI PLL Div[3] drives uart */
-	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
-					    HL_PLL_DIV3);
-
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor;
-	/* As per Goya2 PLL cfg.xlsx, PCI PLL Div[3] drives SPI */
-	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
-					   HL_PLL_DIV3);
-
-	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
-}
 
 /* configure yellow LED(GPIO2) in output mode and turn on(active low) */
 void preboot_led_on(void)
@@ -77,72 +39,6 @@
 	sha256_csum((void *)(uintptr_t)desc->loadaddr, desc->img_size, hash);
 
 	return !memcmp(hash, desc->chksum, ARRAY_SIZE(hash));
-}
-
-void hl_pci_iatu_init(void)
-{
-	struct hl_pci_inbound_region inbound_region;
-	struct hl_pci_outbound_region outbound_region;
-
-	hl_pcie_disable_access();
-
-	/* Inbound Region 0 - Bar - - Point to SRAM and CFG */
-	inbound_region.mode = PCI_BAR_MATCH_MODE;
-	inbound_region.bar = PCI_BAR_ID_0;
-	inbound_region.addr = SDRAM_FULL_BASE;
-	hl_pci_set_inbound_region(0, &inbound_region);
-
-	/* Inbound Region 1 - Bar 4 - Point to DRAM */
-	inbound_region.mode = PCI_BAR_MATCH_MODE;
-	inbound_region.bar = PCI_BAR_ID_4;
-	inbound_region.addr = DRAM_BASE;
-	hl_pci_set_inbound_region(1, &inbound_region);
-
-	/* Outbound Region 0 - Point to Host */
-	outbound_region.addr = HOST_PHYS_BASE;
-	outbound_region.size = HOST_PHYS_SIZE;
-	hl_pci_set_outbound_region(&outbound_region);
-
-	hl_pcie_enable_access();
-}
-
-void lkd_dev_spec_regs(struct cpu_dyn_regs *regs)
-{
-	regs->cpu_pq_base_addr_low = mmCPU_IF_PQ_BASE_ADDR_LOW;
-	regs->cpu_pq_base_addr_high = mmCPU_IF_PQ_BASE_ADDR_HIGH;
-	regs->cpu_pq_length = mmCPU_IF_PQ_LENGTH;
-	regs->cpu_pq_init_status = mmCPU_IF_QUEUE_INIT;
-	regs->cpu_eq_base_addr_low = mmCPU_IF_EQ_BASE_ADDR_LOW;
-	regs->cpu_eq_base_addr_high = mmCPU_IF_EQ_BASE_ADDR_HIGH;
-	regs->cpu_eq_length = mmCPU_IF_EQ_LENGTH;
-	regs->cpu_eq_ci = mmCPU_IF_EQ_RD_OFFS;
-	regs->cpu_cq_base_addr_low = mmCPU_IF_CQ_BASE_ADDR_LOW;
-	regs->cpu_cq_base_addr_high = mmCPU_IF_CQ_BASE_ADDR_HIGH;
-	regs->cpu_cq_length = mmCPU_IF_CQ_LENGTH;
-	regs->cpu_pf_pq_pi = mmCPU_IF_PF_PQ_PI;
-	regs->cpu_boot_dev_sts0 = mmCPU_BOOT_DEV_STS0;
-	regs->cpu_boot_dev_sts1 = mmCPU_BOOT_DEV_STS1;
-	regs->cpu_boot_err0 = mmCPU_BOOT_ERR0;
-	regs->cpu_boot_err1 = mmCPU_BOOT_ERR1;
-	regs->cpu_boot_status = mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
-	regs->fw_upd_sts = mmUPD_STS;
-	regs->fw_upd_cmd = mmUPD_CMD;
-	regs->fw_upd_pending_sts = mmUPD_PENDING_STS;
-	regs->fuse_ver_offset = mmFUSE_VER_OFFSET;
-	regs->preboot_ver_offset = mmPREBOOT_VER_OFFSET;
-	regs->uboot_ver_offset = mmUBOOT_VER_OFFSET;
-	regs->hw_state = mmHW_STATE;
-	regs->kmd_msg_to_cpu = mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU;
-	regs->cpu_cmd_status_to_host = mmCPU_CMD_STATUS_TO_HOST;
-	regs->gic_tpc_qm_irq_ctrl = mmGIC_TPC_QM_IRQ_CTRL_POLL_REG;
-	regs->gic_mme_qm_irq_ctrl = mmGIC_MME_QM_IRQ_CTRL_POLL_REG;
-	regs->gic_dma_qm_irq_ctrl = mmGIC_DMA_QM_IRQ_CTRL_POLL_REG;
-	regs->gic_rot_qm_irq_ctrl = mmGIC_ROT_QM_IRQ_CTRL_POLL_REG;
-	regs->gic_dma_core_irq_ctrl = mmGIC_DMA_CR_IRQ_CTRL_POLL_REG;
-	regs->gic_host_pi_upd_irq = mmGIC_HOST_PI_UPD_IRQ_POLL_REG;
-	regs->gic_host_halt_irq = mmGIC_HOST_HALT_IRQ_POLL_REG;
-	regs->gic_host_ints_irq  = mmGIC_HOST_INTS_IRQ_POLL_REG;
-	regs->gic_host_soft_rst_irq = mmGIC_HOST_SOFT_RST_IRQ_POLL_REG;
 }
 
 /*
diff --git a/arch/greco/common/asic.c b/arch/greco/common/asic.c
index 2b0934c..2c586b4 100644
--- a/arch/greco/common/asic.c
+++ b/arch/greco/common/asic.c
@@ -5,4 +5,114 @@
  *
  */
 
-/* TODO: Greco asic(HL_TARGET) specific common functions to be moved here */
+
+#include <psoc_global_conf_regs.h>
+#include <cpu_if_regs.h>
+
+#include <bitops.h>
+#include <preboot.h>
+#include <uart.h>
+#include <pll.h>
+#include <hl_boot_if.h>
+#include <pci.h>
+
+u32 hl_skip_pcie_init(void)
+{
+	/* Use DumpSel[2] to skip PCIe initialization */
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) & BIT(10);
+}
+
+u64 hl_pll_timestamp_freq_get(void)
+{
+	/* As per Goya2 PLL cfg.xslx, PCI PLL Div[3] drives CPU timestamp */
+	return hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE, HL_PLL_DIV3);
+}
+
+void hl_uart_init(void)
+{
+	/* As per Goya2 PLL cfg.xlsx, PCI PLL Div[3] drives uart */
+	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
+					    HL_PLL_DIV3);
+
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
+	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+	/* As per Goya2 PLL cfg.xlsx, PCI PLL Div[3] drives SPI */
+	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
+					   HL_PLL_DIV3);
+
+	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	return factor > 2 ? factor : 2;
+}
+
+void hl_pci_iatu_init(void)
+{
+	struct hl_pci_inbound_region inbound_region;
+	struct hl_pci_outbound_region outbound_region;
+
+	hl_pcie_disable_access();
+
+	/* Inbound Region 0 - Bar - - Point to SRAM and CFG */
+	inbound_region.mode = PCI_BAR_MATCH_MODE;
+	inbound_region.bar = PCI_BAR_ID_0;
+	inbound_region.addr = SDRAM_FULL_BASE;
+	hl_pci_set_inbound_region(0, &inbound_region);
+
+	/* Inbound Region 1 - Bar 4 - Point to DRAM */
+	inbound_region.mode = PCI_BAR_MATCH_MODE;
+	inbound_region.bar = PCI_BAR_ID_4;
+	inbound_region.addr = DRAM_BASE;
+	hl_pci_set_inbound_region(1, &inbound_region);
+
+	/* Outbound Region 0 - Point to Host */
+	outbound_region.addr = HOST_PHYS_BASE;
+	outbound_region.size = HOST_PHYS_SIZE;
+	hl_pci_set_outbound_region(&outbound_region);
+
+	hl_pcie_enable_access();
+}
+
+void lkd_dev_spec_regs(struct cpu_dyn_regs *regs)
+{
+	regs->cpu_pq_base_addr_low = mmCPU_IF_PQ_BASE_ADDR_LOW;
+	regs->cpu_pq_base_addr_high = mmCPU_IF_PQ_BASE_ADDR_HIGH;
+	regs->cpu_pq_length = mmCPU_IF_PQ_LENGTH;
+	regs->cpu_pq_init_status = mmCPU_IF_QUEUE_INIT;
+	regs->cpu_eq_base_addr_low = mmCPU_IF_EQ_BASE_ADDR_LOW;
+	regs->cpu_eq_base_addr_high = mmCPU_IF_EQ_BASE_ADDR_HIGH;
+	regs->cpu_eq_length = mmCPU_IF_EQ_LENGTH;
+	regs->cpu_eq_ci = mmCPU_IF_EQ_RD_OFFS;
+	regs->cpu_cq_base_addr_low = mmCPU_IF_CQ_BASE_ADDR_LOW;
+	regs->cpu_cq_base_addr_high = mmCPU_IF_CQ_BASE_ADDR_HIGH;
+	regs->cpu_cq_length = mmCPU_IF_CQ_LENGTH;
+	regs->cpu_pf_pq_pi = mmCPU_IF_PF_PQ_PI;
+	regs->cpu_boot_dev_sts0 = mmCPU_BOOT_DEV_STS0;
+	regs->cpu_boot_dev_sts1 = mmCPU_BOOT_DEV_STS1;
+	regs->cpu_boot_err0 = mmCPU_BOOT_ERR0;
+	regs->cpu_boot_err1 = mmCPU_BOOT_ERR1;
+	regs->cpu_boot_status = mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
+	regs->fw_upd_sts = mmUPD_STS;
+	regs->fw_upd_cmd = mmUPD_CMD;
+	regs->fw_upd_pending_sts = mmUPD_PENDING_STS;
+	regs->fuse_ver_offset = mmFUSE_VER_OFFSET;
+	regs->preboot_ver_offset = mmPREBOOT_VER_OFFSET;
+	regs->uboot_ver_offset = mmUBOOT_VER_OFFSET;
+	regs->hw_state = mmHW_STATE;
+	regs->kmd_msg_to_cpu = mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU;
+	regs->cpu_cmd_status_to_host = mmCPU_CMD_STATUS_TO_HOST;
+	regs->gic_tpc_qm_irq_ctrl = mmGIC_TPC_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_mme_qm_irq_ctrl = mmGIC_MME_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_dma_qm_irq_ctrl = mmGIC_DMA_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_rot_qm_irq_ctrl = mmGIC_ROT_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_dma_core_irq_ctrl = mmGIC_DMA_CR_IRQ_CTRL_POLL_REG;
+	regs->gic_host_pi_upd_irq = mmGIC_HOST_PI_UPD_IRQ_POLL_REG;
+	regs->gic_host_halt_irq = mmGIC_HOST_HALT_IRQ_POLL_REG;
+	regs->gic_host_ints_irq  = mmGIC_HOST_INTS_IRQ_POLL_REG;
+	regs->gic_host_soft_rst_irq = mmGIC_HOST_SOFT_RST_IRQ_POLL_REG;
+}
